### PR TITLE
Dependabot can update kubernetes dependencies as a group

### PR DESCRIPTION
**Pull request by kannon92**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Dependabot is opening up PRs individually for k8s packages. We can actually group all the Kubernetes dependencies together to minimize the amount of PRs that we are getting for updates.

See https://github.com/cri-o/cri-o/pull/7707 for an example of what this syntax does.
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -9,6 +9,11 @@ updates:
     labels:
       - "ok-to-test"
       - "release-note-none"
+    groups:
+      kubernetes:
+        patterns:
+          - "k8s.io/*"
+          - "sigs.k8s.io/*"
     ignore:
```
Comment (trasc) [inline .github/dependabot.yml:17]: Should we also update the ignore section? I'm not very familiar with dependabot, but maybe we at least need to have the same policy for `sigs.ks8.io` ...

cc: @alculquicondor

Comment (kannon92) [reply .github/dependabot.yml:17]: Looking at this, do we see any value in having all sig projects updated together? I see that this repo tends to have very different projects in the sig group so I think individual updates is fine. 

Ignore is mostly about doing updates via patches only.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -9,6 +9,11 @@ updates:
     labels:
       - "ok-to-test"
       - "release-note-none"
+    groups:
+      kubernetes:
+        patterns:
+          - "k8s.io/*"
+          - "sigs.k8s.io/*"
```
Comment (alculquicondor) [inline .github/dependabot.yml:16]: I don't think sigs should be part of the same group.

Comment (kannon92) [reply .github/dependabot.yml]: I agree. Pushed up a fix for that.

**Pull request merged**